### PR TITLE
Bookstore: create and remove books

### PR DESCRIPTION
**Pull request by NasKhalil**
This branch includes the following features implemented:

- configuring the Redux Store.
- book's `actions` and `reducer`.
- implement `addNewBook` function.
- implement `deleteBook`function.

### CODE REVIEW

File: src/components/booksComponent/AddBookForm.js [hunk 1]
```diff
@@ -1,14 +1,65 @@
...
-      <input type="text" placeholder="Categorie" />
-      <button type="submit">Add Book</button>
-    </form>
-  </div>
-);
+import { React, useState } from 'react';
+import PropTypes from 'prop-types';
+
+// import { useDispatch } from 'react-redux';
+// import { addBook } from '../../redux/books/books';
```
Comment (Lameck1) [inline src/components/booksComponent/AddBookForm.js]: - [x] Please delete any part of code that's not in use and is commented out, they are several in this particular file. It makes the codebase dirty.

File: src/components/booksComponent/BooksItem.js [hunk 2]
```diff
@@ -1,11 +1,29 @@
...
+    <div>
+      <h2>{title}</h2>
+      <p>{author}</p>
+      <button onClick={deleteBook} data-bookid={id} type="button">Delete</button>
+    </div>
+  );
+};
+
+BooksItem.propTypes = {
+  id: PropTypes.number.isRequired,
```
Comment (Lameck1) [inline src/components/booksComponent/BooksItem.js]: - [x] This PropType definition presents us with the error below. As you can see from the screenshot, the id is actually a string when a book is created. Kindly work on this.

  image

**Review by Lameck1 — CHANGES_REQUESTED** [review]
Hi, @NasKhalil, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @NasKhalil,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**